src/main/app/windowManager.js

```diff
@@ -197,10 +220,73 @@ class WindowManager extends EventEmitter {
     return result
   }
 
+  /**
+   * Find the best window to open the files in.
+   *
+   * @param {string[]} fileList File full paths.
+   * @returns {{windowId: string, fileList: string[]}[]} An array of files mapped to a window id or null to open in a new window.
+   */
+  findBestWindowToOpenIn (fileList) {
+    if (!fileList || !Array.isArray(fileList) || !fileList.length) return []
+    const { windows } = this
+    const lastActiveEditorId = this.getActiveEditorId() // editor id or null
+
+    if (this.windowCount <= 1) {
+      return [ { windowId: lastActiveEditorId, fileList } ]
+    }
+
+    // Array of scores, same order like fileList.
+    let best = null
```

Comment:
Also need to rename `best`, `best` usually makes people feel like one instance, not an array.